edgelet/iotedge/src/check/checks/container_engine_is_moby.rs

```diff
@@ -75,10 +75,10 @@ impl ContainerEngineIsMoby {
         // Older releases of Moby do not identify themselves in any unique way. Moby devs recommended assuming that anything less than version 10 is Moby,
         // since these old releases are v3.x and regular Docker is in the high 10s.
         //
-        // Newer releases of Moby follow Docker CE versioning, but have a "+azure" suffix, eg "19.03.12+azure"
+        // Newer releases of Moby follow Docker CE versioning, but have a "+azure" substring, eg "19.03.12+azure"
         //
-        // Therefore Docker CE is anything with major version >= 10 but without a "+azure" suffix.
-        if docker_server_major_version >= 10 && !docker_server_version.ends_with("+azure") {
+        // Therefore Docker CE is anything with major version >= 10 but without a "+azure" substring.
```

Comment:
Have we confirmed with Moby on versioning naming conventions?

Reply:
Do you have a link?

Reply:
Is there a reason to worry about the `+azure` at all?

Reply:
The reason we have this check is that we want to enforce people are using Moby, not regular Docker, since we only provide support officially for users using Moby.

Reply:
Thanks!

Reply:
After discussion with @cpuguy83 , this method works, however it might not be the best solution if schema changes again in the future.
I am merging it now because it is better than the current situation but probably we need to find a better long term solution. I'll create a PBI.

We could consider checking at package install?

Reply:
>We could consider checking at package install?

No, because we don't have the iotedge package depend on moby, which in turn is because we want people to be able to use regular Docker for development. Moby is only a requirement for production. So it has to be enforced at runtime when the user is evaluating production readiness.